shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -473,6 +525,18 @@ - (void)dispatchMouseEvent:(nonnull NSEvent*)event {
   [self dispatchMouseEvent:event phase:phase];
 }
 
+- (void)dispatchGestureEvent:(nonnull NSEvent*)event {
+  if (event.phase == NSEventPhaseBegan || event.phase == NSEventPhaseMayBegin) {
+    [self dispatchMouseEvent:event phase:kPanZoomStart];
+  } else if (event.phase == NSEventPhaseChanged) {
+    [self dispatchMouseEvent:event phase:kPanZoomUpdate];
+  } else if (event.phase == NSEventPhaseEnded || event.phase == NSEventPhaseCancelled) {
+    [self dispatchMouseEvent:event phase:kPanZoomEnd];
+  } else if (event.phase == NSEventPhaseNone && event.momentumPhase == NSEventPhaseNone) {
+    [self dispatchMouseEvent:event phase:kHover];
+  }
```

Comment:
Maybe assert otherwise?

Reply:
Some events are skipped for good reason (system-generated momentum). I added an assert that would handle any unexpected events though.